src/Okta.Idx.Sdk/IdxClientExtensions.cs

```diff
@@ -0,0 +1,54 @@
+﻿// <copyright file="IdxClientExtensions.cs" company="Okta, Inc">
+// Copyright (c) 2020 - present Okta, Inc. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Okta.Idx.Sdk
+{
+    internal static class IdxClientExtensions
+    {
+        internal static async Task<IIdxResponse> ContinueWithRemediationOption(this IIdxResponse response, string remediationType, IdxRequestPayload request, CancellationToken cancellationToken = default)
+        {
+            var remediationOption = response
+                    .Remediation
+                    .RemediationOptions
+                    .FirstOrDefault(x => x.Name == remediationType);
+
+            if (remediationOption == null)
+            {
+                throw new UnexpectedRemediationException(remediationType, response);
+            }
+
+            return await remediationOption
+                    .ProceedAsync(request, cancellationToken);
+        }
+
+        internal static async Task<IIdxResponse> ContinueWithFirstFoundRemediationOption(this IIdxResponse response, IList<string> remediationTypeNames, IdxRequestPayload request, CancellationToken cancellationToken = default)
```

Comment:
I'm not sure about using this method, since from my code I want more control of the remediation that I'm executing. If I don't know what remediation was executed then I won't know how to process the response. Does it make sense to you?

Reply:
I did not know the exact use-case for this method. If we return the remediation option that was found and processed, will it help?

Reply:
We can return it in an out-parameter for example

Reply:
We cannot return it in out param, because it's an async function, we can have another complex return type with the `RemediationOptionName` and `IdxResponse`

Reply:
Are we using this somewhere? If not, I prefer to remove this one.